src/dotnet/Notification.UI.Blazor/notification-ui.ts

```diff
@@ -0,0 +1,175 @@
+import { initializeApp } from 'firebase/app';
+import { getMessaging, getToken, GetTokenOptions, onMessage } from 'firebase/messaging';
+import { Log, LogLevel, LogScope } from 'logging';
+import { AppKind } from '../UI.Blazor/Services/BrowserInfo/browser-info';
+
+const LogScope: LogScope = 'NotificationUI';
+const debugLog = Log.get(LogScope, LogLevel.Debug);
+const infoLog = Log.get(LogScope, LogLevel.Info);
+const warnLog = Log.get(LogScope, LogLevel.Warn);
+const errorLog = Log.get(LogScope, LogLevel.Error);
+
+export class NotificationUI {
+    private static baseLayoutRef?: DotNet.DotNetObject = null;
+    private static appKind?: AppKind = null;
+
+    public static async init(baseLayoutRef: DotNet.DotNetObject, appKind: AppKind): Promise<void> {
+        // probably init can be called multiple times on MAUI
+        debugLog?.log('init');
+        NotificationUI.baseLayoutRef = baseLayoutRef;
+        NotificationUI.appKind = appKind;
+
+        if (appKind === 'Maui')
+            return;
+
+        const status = await NotificationUI.getNotificationPermissionStatus();
+        await NotificationUI.updateNotificationStatus(status);
+        NotificationUI.registerNotificationHandler();
```

Comment:
await?

Reply:
It's void and doesn't contain async statements